arborcat.module

```diff
@@ -1,120 +1,216 @@
+          $result = curl_exec($ch);
+          if ($result != null) {
+              preg_match_all('%name="web_patroninserted_door.*?value="([\d]+?)"%s', $result, $matches);
+              if ($matches[1][0]>0) {
+                  $available = true;
+              }
+          }
       }
-        else if($locker_server=='mcblockers'){
-        $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, \Drupal::config('arborcat.settings')->get('mcb_lockers'));
-        curl_setopt($ch, CURLOPT_HEADER, 0);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-        curl_setopt($ch, CURLOPT_USERPWD, \Drupal::config('arborcat.settings')->get('lockers_pass'));
-        $result = curl_exec($ch);
-        preg_match_all('%name="web_patroninserted_door.*?value="([\d]+?)"%s', $result, $matches);
-        if ($matches[1][0]>0) {
-          $available = TRUE;
-        }
+      return $available;
+  }
+
+  /*
+   * returns a list of request pickup locations.
+   * If a branch location is passed in, that a filtered list of pickup locations is returned
+   */
+  function arborcat_pickupLocations($destLocation = null)
+  {
+      $db = \Drupal::database();
+      $query = $db->select('arborcat_pickup_location', 'apl');
+      $query->fields('apl', ['locationId', 'branchLocationId', 'timePeriod', 'timePeriodStart', 'timePeriodEnd', 'maxLockers', 'locationName', 'locationDescription']);
+        
+      // add in a condition if a location is supplied to filter on
+      if (3 == strlen($destLocation)) {
+          $query->condition('apl.branchLocationId', (int) $destLocation, '=');
       }
-    return $available;
+      $result = $query->execute();
+      
+      $pickupLocationRecords = $result->fetchAll();
+      return $pickupLocationRecords;
+  }
+
+   /*
+   * Checks the availability of a locker fo the specified date and timePeriod.
+   * If a branch location is passed in, that a filtered list of pickup locations is returned
+   */
+  function lockerAvailableForDateAndTimeSlot($queryDate, $locationObject)
+  {
+      $availableLockers = 0;
+      // Query current arborcat_patron_pickup_request records for the date and timePeriod to get inuse count
+      // fields pickupDate & timeSlot
+      $db = \Drupal::database();
+      $timeperiod = $locationObject->timePeriod;
+      $query = $db->select('arborcat_patron_pickup_request', 'appr');
+      $query-> fields('appr', ['id']);
+      $query-> condition('pickupDate', $queryDate);
+      $query-> condition('timeSlot', $timeperiod, '=');
+      $results = $query->execute()->fetchAll();
+
+      $lockersInUse = count($results);
+      $availableLockers = $locationObject->maxLockers - $lockersInUse;
+      
+      return ($availableLockers > 0);
   }
-  
-?>
+
+function addPickupRequest($pickupLocation, $pickupDay, $timeSlot, $contactEmail, $contactPhone, $contactSMS)
+{
+    $result = [];
+    
+    $user = \Drupal\user\Entity\User::load(\Drupal::currentUser()->id());
```

Comment:
Should be able to do a authentication check on the route definition rather than doing another explicit one here, just a matter of finding the right permission name